### PR TITLE
Small optimizations

### PR DESCRIPTION
**Pull request by MarkCiliaVincenti**

### CODE REVIEW

File: src/DotNet.RateLimiter/ServiceCollectionExtension.cs [hunk 1]
```diff
@@ -49,11 +49,7 @@ public static void AddRateLimitService(this IServiceCollection services, IConfig
             else
             {
                 services.AddMemoryCache();
-                services.AddSingleton(new AsyncKeyedLocker<string>(o =>
-                {
-                    o.PoolSize = 20;
```
Comment (sa-es-ir) [inline src/DotNet.RateLimiter/ServiceCollectionExtension.cs:54]: Is the ``PoolSize`` handled internally in the new AsyncLock version?

Comment (MarkCiliaVincenti) [reply src/DotNet.RateLimiter/ServiceCollectionExtension.cs:54]: Yes it is. You can either handle it yourself or rely on the defaults. Currently I feel this combination is the best, but perhaps I will one day find a more optimal combination and change the defaults. We can revert this part if you want to retain control.

Comment (sa-es-ir) [reply src/DotNet.RateLimiter/ServiceCollectionExtension.cs:54]: @MarkCiliaVincenti I think better to keep visibility here, could you please revert this part?

Comment (MarkCiliaVincenti) [reply src/DotNet.RateLimiter/ServiceCollectionExtension.cs]: Done

**Review by sa-es-ir — APPROVED** [review]

**Comment by MarkCiliaVincenti** [comment]
When you write release notes, please make note of the small breaking change. The IP whitelist is no longer a list but a hashset. This helps more efficient matching, but if someone is using this feature, upgrading to the new DotNet.RateLimit NuGet package (whenever it comes out) might end up with code not compiling for them and they might need to add a `.ToHashSet()` or something.

**Pull request merged**